src/call_builder.ts

```diff
@@ -14,29 +13,18 @@ const version = require("../package.json").version;
 // query-param.
 const JOINABLE = ["transaction"];
 
-type Constructable<T> = new (e: string) => T;
-
 export interface EventSourceOptions<T> {
   onmessage?: (value: T) => void;
   onerror?: (event: MessageEvent) => void;
   reconnectTimeout?: number;
 }
 
-let EventSource: Constructable<EventSource>;
 const anyGlobal = global as any;
-
-if (anyGlobal.EventSource) {
-  EventSource = anyGlobal.EventSource;
-} else if (isNode) {
-  /* tslint:disable-next-line:no-var-requires */
-  EventSource = require("eventsource");
-} else if (anyGlobal.window.EventSource) {
-  EventSource = anyGlobal.window.EventSource;
-} else {
-  // require("eventsource") for React Native environment
-  /* tslint:disable-next-line:no-var-requires */
-  EventSource = require("eventsource");
-}
+type Constructable<T> = new (e: string) => T;
+// require("eventsource") for Node and React Native environment
+let EventSource: Constructable<EventSource> = anyGlobal.EventSource ??
+  anyGlobal.window?.EventSource ??
+  require("eventsource");
```

Comment:
The old `isNode` package was checking to see if `process` is defined instead of checking if `window` is defined (which it looks like this solution is doing).

I've run into an issue in Freighter where I've tried to run stellar sdk in a service worker and because `window` wasn't defined, it falsely assumed I was in a node environment, which has led to some issues.

Reply:
Here's the PR that addressed the mentioned Freighter issue: https://github.com/stellar/js-stellar-base/pull/567

It might not actually matter here, but thought I would mention

Reply:
Good context, thanks! You're right that we're doing a window check, but I think it should still be okay here because the default logic flow stays the same. Can you double-check my rationale?

Previously:

1. is there an `EventSource` in the `global` variable? use it
2. is there a `process` (via `isNode`)? then `require()` it
3. is there a `window` in the global variable AND an `EventSource` on that `window`? use it
4. default to trying to `require()` it

Now, we drop (2) and move it to the "default case." I think this is fine, because if we're in Node, there won't be a `window`, and if there **is** a window, we can still use its `EventSource` if it has one. Then if all else fails, we try importing it.